doc/development.md

```diff
@@ -120,3 +120,180 @@
+
+Continuation lines should align wrapped elements either vertically using
+Python's implicit line joining inside parentheses, brackets and braces, or
+using a hanging indent. When using a hanging indent, the following should
+be considered:
+- There should be no arguments on the first line.
+- Further indentation should be used to clearly distinguish itself as a continuation line.
+- Hanging indentation should stick to the 4-space rule.
+```python
+# Add 4 spaces (an extra level of indentation) to distinguish arguments from the rest.
+# Consider that having too many arguments in a function is
+foo = long_function_name(
+    var_one, var_two, var_three, var_four)
+
+# Recommended extra indentation with multiline if-statements.
+if (condition_1 and condition_2 [...]
+        and condition_n):
+    do_something()
+
+# But this is also possible.
+def long_function_name(
+        var_one, var_two, var_three,
+        var_four):
+    print(var_one)
+```
+
+To break comprehension Lists, Dicts, Sets as well as generator expressions,
+keep expressions that would be written in a single line together.
+Continuation lines stick to the same rules as above. Closing
+brace/bracket/parenthesis shall line up under the first character of the
+line start.
+
+```python
+# Vertically aligned and extra 4 spaces for condition continuation
+foo = [var for var in bar_list
+       if condition_1
+           and condition_2
+           and condition_3]
+```
+
+### Docstrings
+
+Always use the three double-quote `"""` format for docstrings 
+(per [PEP 257](https://peps.python.org/pep-0257/)).
+A docstring shall be organized as multiple summary lines (one physical line
+not exceeding 79 characters) terminated by a period. First and last line
+of the docstring shall be blank.
+
+Module functions and classes should have a docstring, unless:
+- Not public.
+- Pertains to a test module.
+
+Class methods do not require docstring, but is recommended when it is a public
+entity or have a complex interface.
+
+The docstring should be imperative-style (e.g., `Verify configuration file.`)
+rather than descriptive-style (e.g., `Verifies configuration file.`).
+The docstring should describe the function's main purpose, its calling syntax,
+and/or semantics, but avoid implementation details.
+
+Certain aspects of a function shall be documented in special sections,
+listed below.
+
+#### Args: (or Arguments:)
+
+List each parameter by name. It is strongly encouraged that the type of the
+parameter follows the name, either enclosed in parentheses, with clear wording,
+or in [type annotation](https://docs.python.org/3/library/typing.html) format).
+Then the description of the parameter
+separated by a colon. If the type is optional or can take multiple types,
+that can be specified with comma-separated lists or use
+type annotation syntax.
+If the description of the parameter is too long to
+fit in 79 characters, following hanging lines are indented with 4 extra
+spaces.
```

Comment:
Are the 4 spaces mandated by something generally accepted or suggested just here?  I find 4, rather than 2, spaces excessive, both visually and as for taking from the available line space.

Reply:
https://developers.google.com/edu/python/introduction?csw=1#indentation

Basically this: 4 spaces is widely accepted in the python community + is recommended by the PEP 8 style guide. But there are other options that use 2 spaces instead, and it is equally fine.

IMO whatever we decide, it should be mandated to keep the code consistent. Inconsistent indentation from within the same project is evil. I'd vote for 4 spaces, as is already the most commonly indentation used in the project files. But I'm open to debate.

Reply:
Ah, you meant only the documentation (my bad). Here we can be more flexible, I assume. I don't have a strong opinion. I can relax it to a recommendation and comment that 2 spaces are also ok (as long as it is consistent with surrounding code).

Reply:
> I can relax it to a recommendation and comment that 2 spaces are also ok 

OK.

> (as long as it is consistent with surrounding code).

... surrounding docstrings.